dfsclient/src/main/java/org/hpccsystems/dfs/client/FileUtility.java

```diff
@@ -0,0 +1,1587 @@
+                    return;
+                }
+
+                Runnable[] tasks = null;
+                try
+                {
+                    tasks = createNonRedistributingCopyTasks(file, createResult, context);
+                }
+                catch (Exception e)
+                {
+                    context.addError("Error while attempting to create copy tasks for file: '" + srcFile + "': " + e.getMessage());
+                    return;
+                }
+
+                try
+                {
+                    executeTasks(tasks, numThreads);
+                }
+                catch (Exception e)
+                {
+                    context.addError("Error while attempting to execute copy tasks for file: '" + srcFile + "': " + e.getMessage());
+                    return;
+                }
+
+                if (context.hasError())
+                {
+                    return;
+                }
+
+                try
+                {
+                    long bytesWritten = context.bytesWritten.get();
+                    long recordsWritten = context.recordsWritten.get();
+                    dfuClient.publishFile(createResult.getFileID(), eclRecordDefn, recordsWritten, bytesWritten, true);
+                }
+                catch (Exception e)
+                {
+                    context.addError("Error while attempting to publish file: '" + destFile + "': " + e.getMessage());
+                    return;
+                }
+            }
+            else
+            {
+                String readArgs[] = {"-read", srcFile, "-url", srcURL,
+                                "-format", "thor", "-user", user, "-pass", pass,
+                                "-out", "tmp-read"};
+
+                performRead(readArgs, context);
+
+                String writeArgs[] = {"-write", "tmp-read" + File.separator +  srcFile.replace(':', '_') + "*" +  " " + destFile,
+                                "-url", srcURL, "-dest_url", destURL,
+                                "-dest_cluster", destClusterName,
+                                "-user", user, "-pass", pass };
+
+                performWrite(writeArgs, context);
+            }
+
+            context.endOperation();
+        }
+    }
+
+    private static void performWrite(String[] args, TaskContext context)
+    {
+        Options options = getWriteOptions();
+        CommandLineParser parser = new DefaultParser();
+
+        CommandLine cmd = null;
+        try
+        {
+            cmd = parser.parse(options, args);
+        }
+        catch (ParseException e)
+        {
+            System.out.println("Error parsing commandline options:\n" + e.getMessage());
+            return;
+        }
+
+        int numThreads = 4;
+        String numThreadsStr = cmd.getOptionValue("num_threads", "" + numThreads);
+        try
+        {
+            numThreads = Integer.parseInt(numThreadsStr);
+        }
+        catch(Exception e) {}

```

Comment:
it feels like this should be an error, at least we should log the reason why we're defaulting to 4 threads